### PR TITLE
WIP: Add validation pass for conversion instructions

### PR DESCRIPTION
**Pull request by atgoo**
The pass checks correctness of operands of instruction in opcode range
OpConvertFToU - OpBitset.

### CODE REVIEW

**Comment by atgoo** [comment]
DO NOT SUBMIT. Fails Vulkan tests.

**Comment by dneto0** [comment]
Thanks for checking on Vulkan CTS shaders.  

I'll mark as WIP for now so I don't look at this for the next while.
Please update this bug and remove the WIP once those shaders have been fixed.

**Comment by atgoo** [comment]
Temporarily disable validation rules for OpUConvert.

**Comment by atgoo** [comment]
Passes VulkanCTS with disabled UConvert check.

File: source/validate_conversion.cpp [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+// Validates correctness of conversion instructions.
+spv_result_t ConversionPass(ValidationState_t& _,
+                            const spv_parsed_instruction_t* inst) {
+  const SpvOp opcode = static_cast<SpvOp>(inst->opcode);
+  const uint32_t result_type = inst->type_id;
+
+  switch (opcode) {
+    case SpvOpConvertFToU: {
+      if (!_.IsUnsignedIntScalarType(result_type) &&
```
Comment (dneto0) [inline source/validate_conversion.cpp]: FYI.  LLVM has isFPOrFPVectorType and isIntOrIntVectorType.  That's a useful idiom we might want to support.

(Not for this CL)

**Comment by dneto0** [comment]
This was rebased, squashed and pushed into master as 39e25fd8

**Pull request closed without merging**